src/main/kotlin/com/varabyte/kobweb/intellij/colors/KobwebColorProvider.kt

```diff
@@ -0,0 +1,222 @@
+@file:Suppress("UseJBColor")
+
+package com.varabyte.kobweb.intellij.colors
+
+import com.intellij.ide.util.PsiNavigationSupport
+import com.intellij.openapi.editor.ElementColorProvider
+import com.intellij.openapi.fileEditor.OpenFileDescriptor
+import com.intellij.psi.PsiElement
+import com.intellij.psi.impl.source.tree.LeafPsiElement
+import com.intellij.psi.util.PsiUtilBase
+import org.jetbrains.kotlin.analysis.api.analyze
+import org.jetbrains.kotlin.analysis.api.components.KtConstantEvaluationMode
+import org.jetbrains.kotlin.idea.base.psi.kotlinFqName
+import org.jetbrains.kotlin.idea.core.util.toPsiFile
+import org.jetbrains.kotlin.idea.references.mainReference
+import org.jetbrains.kotlin.js.translate.declaration.hasCustomGetter
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.*
+import java.awt.Color
+import kotlin.math.abs
+
+// This prevents the color tracer from checking the codebase ridiculously deep,
+// which might cause lag or even worse a stackoverflow
+private const val MAX_SEARCH_DEPTH = 15
+
+private const val KOBWEB_COLOR_COMPANION_FQ_NAME = "com.varabyte.kobweb.compose.ui.graphics.Color.Companion"
+
+/**
+ * Enables showing those small rectangular color previews next to the line number
+ */
+class KobwebColorProvider : ElementColorProvider {
+
+    override fun getColorFrom(element: PsiElement): Color? = when {
+        element !is LeafPsiElement -> null
+        element.elementType != KtTokens.IDENTIFIER -> null
+        element.parent is KtProperty -> null // Avoid showing multiple previews
+        else -> traceColor(element.parent) // Leaf is just text. The parent is the actual object
+    }
+
+    /**
+     * Setting colors is not (yet) supported
+     */
+    override fun setColorTo(element: PsiElement, color: Color) = Unit
+}
+
+/**
+ * Tries resolving references as deep as possible and checks if a Kobweb color is being referred to
```

Comment:
Optional:

It can be a good idea to document nullable values. Here, it's not really necessary, it's pretty obvious from context. But I'm mentioning it just in case you haven't heard this advice before.

For this method, doing so would look like this:

```
Tries resolving references as deep as possible and checks if a Kobweb color is being referred to.

@return the color being referenced, or null if the [element] ultimately doesn't resolve to a color at all (which is common) or if it did but the amount of times we'd have to follow references to get to the color at the bottom is too many, or it *was* a color but not one we could extract specific information about (e.g. a method that returns one of two colors based on a condition).
```

Up to you to determine if you think this information is worth surfacing in this case.